src/zerocash/utils/util.cpp

```diff
@@ -203,68 +203,65 @@ void concatenateVectors(const std::vector<unsigned char>& A, const std::vector<u
 }
 
 void sha256(const unsigned char* input, unsigned char* hash, int len) {
-	SHA256_CTX_mod ctx256;
-
-	sha256_init(&ctx256);
-	sha256_update(&ctx256, input, len);
-	sha256_final_no_padding(&ctx256, hash);
+	CSHA256 hasher;
+    hasher.Write(input, len);
+    hasher.FinalizeNoPadding(hash);
 }
 
-void sha256(SHA256_CTX_mod* ctx256, const unsigned char* input, unsigned char* hash, int len) {
-	sha256_init(ctx256);
-	sha256_update(ctx256, input, len);
-	sha256_final_no_padding(ctx256, hash);
+void sha256(CSHA256& hasher, const unsigned char* input, unsigned char* hash, int len) {
+	hasher.Write(input, len);
+    hasher.FinalizeNoPadding(hash);
```

Comment:
The old behavior was to re-initialize the context. The new behavior doesn't do that. Does that matter?